solr/solrj/src/test/org/apache/solr/client/solrj/impl/ConcurrentUpdateSolrClientTest.java

```diff
@@ -346,4 +347,25 @@ public OutcomeCountingConcurrentUpdateSolrClient build() {
       }
     }
   }
+
+  /**
+   * Test that connection timeout information is passed to the HttpSolrClient that handle non add operations
+   */
+  @Test(timeout = 1000)
+  public void testConnectionTimeoutOnCommit() throws IOException, SolrServerException {
+    // 240.0.0.0/4 is reserved for future use by the IANA and is in effect a black hole
+    try (ConcurrentUpdateSolrClient client = new ConcurrentUpdateSolrClient.Builder("http://240.0.0.1:8983/")
```

Comment:
Please use `SolrTestCaseJ4.DEAD_HOST` instead of a real networking address so that it fails fast. After SOLR-14050, the security policy for tests allows connections to loopback only.

Reply:
You are right, this test fails on current master. (This PR is quite ancient.) 

Unfortunately, DEAD_HOST does not work either. For this test to actually test the right condition, the connection needs to time out and not fail on connection. 

I have updated the pull request with another solution that works within current security policy and fails quite fast (1 ms).